packages/application-shell-connectors/src/components/application-context/normalizers.ts

```diff
@@ -0,0 +1,218 @@
+ *       canViewOrders: { values: ['canada'] },
+ *     }
+ *   }
+ * }
+ */
+type NormalizedStoresGroupByResourceType = {
+  [key: string]: TStoreDataFence[];
+};
+
+const normalizeAppliedDataFencesForStoresByResourceType = (
+  dataFences: Maybe<TStoreDataFence>[]
+) => {
+  const groupedByResourceType = dataFences.reduce<
+    NormalizedStoresGroupByResourceType
+  >((previousGroupsOfSameType, appliedDataFence) => {
+    if (!appliedDataFence) return previousGroupsOfSameType;
+    const previousGroup = previousGroupsOfSameType[appliedDataFence.group];
+    return {
+      ...previousGroupsOfSameType,
+      [appliedDataFence.group]: [...(previousGroup || []), appliedDataFence],
+    };
+  }, {});
+  return Object.entries(groupedByResourceType).reduce<
+    NormalizedGroupedByResourceType
+  >((previousGroupedByResourceType, [resourceType, dataFences]) => {
+    const groupByDataFenceName = dataFences.reduce<
+      NormalizedGroupedByPermission
+    >((nextDataFenceValues, dataFence) => {
+      const dataFenceByName = nextDataFenceValues[dataFence.name] || {
+        values: [],
+      };
+      return {
+        ...nextDataFenceValues,
+        [dataFence.name]: {
+          ...dataFenceByName,
+          values: [...dataFenceByName.values, dataFence.value],
+        },
+      };
+    }, {});
+    return {
+      ...previousGroupedByResourceType,
+      [resourceType]: groupByDataFenceName,
+    };
+  }, {});
+};
+
+// input:
+// [
+//   {
+//     type: 'store',
+//     name: 'canManageOrders',
+//     value: 'usa',
+//     group: 'orders',
+//   },
+//   {
+//     type: 'store',
+//     name: 'canManageOrders',
+//     value: 'germany',
+//     group: 'orders',
+//   },
+//   {
+//     type: 'store',
+//     name: 'canViewOrders',
+//     value: 'canada',
+//     group: 'orders',
+//   },
+// ]
+// output:
+// {
+//   store: {
+//     orders: {
+//       canManageOrders: { values: ['usa', 'germany'] },
+//       canViewOrders: { values: ['canada'] },
+//     }
+//   }
+// }
+
+type NormalizedGroupByType = {
+  [key: string]: Maybe<TStoreDataFence>[];
+};
+
+export const normalizeAllAppliedDataFences = (
+  project: TFetchProjectQuery['project']
+) => {
+  if (!project) return null;
+  const allAppliedDataFences = project.allAppliedDataFences;
+  if (allAppliedDataFences.length === 0) {
+    return null;
+  }
+  const groupedByType = allAppliedDataFences.reduce<NormalizedGroupByType>(
+    (previousGroupsOfSameType, appliedDataFence) => {
+      if (!appliedDataFence) return previousGroupsOfSameType;
+      const previousGroup = previousGroupsOfSameType[appliedDataFence.type];
+      return {
+        ...previousGroupsOfSameType,
+        [appliedDataFence.type]: [...(previousGroup || []), appliedDataFence],
+      };
+    },
+    {}
+  );
+  const normalizedDataFences = {
+    ...('store' in groupedByType
+      ? {
+          store: normalizeAppliedDataFencesForStoresByResourceType(
```

Comment:
NOTE that this function is specific to the `store` type. Therefore, we can use this function to explicitly normalize the store values.
In the future, if we have more data fence types, we can have specific normalize functions for each one of them. In case some share the same format, we can reuse the normalize functions.
However, for now I would prefer to keep it explicit.

Also note that the return type of the `normalizeAllAppliedDataFences` can be `null` or an object optionally containing a `store` key, hence the need for this: https://github.com/commercetools/merchant-center-application-kit/pull/1064/files#r325633637